js/src/scatter.ts

```diff
@@ -335,6 +432,10 @@ class ScatterView extends widgets.WidgetView {
         const current  = new values.Values(scalar_names, [], this.get_current.bind(this), sequence_index, vector4_names);
         const previous = new values.Values(scalar_names, [], this.get_previous.bind(this), sequence_index_previous, vector4_names);
 
+        // Workaround for shader issue - color redefine 
+        current.ensure_array(["color"]);
+        instanced_geo.addAttribute("color_current", new THREE.BufferAttribute(current.array_vec4.color, 4));
```

Comment:
I don't fully understand the solution here, I'm afraid it will upload the color array twice to the GPU now, as 'color' and 'color_current'. I wonder if we should rename all `x` and `x_previous` to `x_next` and `x_previous` to avoid name collisions.

Reply:
I believe the 'ensure_array' line is not longer necessary. I must study the scatter.ts code/behavior more before I give a definite answer.